posts/2023-08-08_study_day/study_day.qmd

```diff
@@ -0,0 +1,84 @@
+---
+title: "It's all relative? - Calculating Relative Days using admiral"
+author:
+  - name: Ben Straub
+description: ""
+date: "2023-08-08"
+# please do not use any non-default categories.
+# You can find the default categories in the repository README.md
+categories: [admiral, ADaMs]
+# feel free to change the image
+image: "admiral.png"
+
+---
+
+<!--------------- typical setup ----------------->
+
+```{r setup, include=FALSE}
+long_slug <- "2023-08-08_study_day"
+# renv::use(lockfile = "renv.lock")
+```
+
+<!--------------- post begins here -----------------> 
+
+Creating `--DY` variables for your ADaMs is super easy using  [`derive_vars_dy()`](https://pharmaverse.github.io/admiral/v0.11.1/reference/derive_vars_dy.html) from the [admiral package](https://pharmaverse.github.io/admiral/cran-release/).
+
+
+Let's build some dummy data with 4 subjects, a start date/time for treatment (`TRTSDTM`), an analysis start date/time variable (`ASTDTM`) and an analysis end date variable (`AENDT`).
+
+```{r, message = FALSE}
+library(admiral)
+library(lubridate)
+library(dplyr)
+
+adam <- tribble(
+  ~USUBJID, ~TRTSDTM, ~ASTDTM, ~AENDT,
+  "001", "2014-01-17T23:59:59", "2014-01-18T13:09:O9", "2014-01-20",
+  "002", "2014-02-25T23:59:59", "2014-03-18T14:09:O9", "2014-03-11",
+  "003", "2014-02-12T23:59:59", "2014-02-18T11:03:O9", "2014-04-17",
+  "004", "2014-03-17T23:59:59", "2014-03-19T13:09:O9", "2014-05-04"
+) %>%
+  mutate(
+    TRTSDTM = as_datetime(TRTSDTM),
+    ASTDTM = as_datetime(ASTDTM),
+    AENDT = ymd(AENDT)
+  )
+```
+
+Okay! Next we run our dataset through `derive_vars_dy()`, specifying the `reference_date` of the variable you want the calculation to be done against and the `source_vars`argument with the variables to help calculate the relative days.  
```

Comment:
This sentence was slightly confusing at a first glance, perhaps change to a bullet-pointed list. Something like

Okay! Next we run our dataset through `derive_vars_dy()`, specifying:
- the reference date of the variable we want the calculation to be done against (`reference_date` argument);
- the variables required to help calculate the relative days  (`source_vars` argument).